src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -83,6 +92,13 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
                 {
                     throw new InvalidOperationException("The monitored collection cannot be the same as the collection storing the leases.");
                 }
+
+                if (attribute.CreateIfNotExists)
+                {
+                    // Not disposing this because it might be reused on other Trigger since Triggers could share lease collection
+                    IDocumentDBService service = _config.GetService(leasesConnectionString);
+                    CreateIfNotExistAsync(service, leaseCollectionLocation.DatabaseName, leaseCollectionLocation.CollectionName, attribute.CollectionThroughput).Wait();
```

Comment:
You can make it `async` and change the return to `return new CosmosDBTriggerBinding(parameter, documentCollectionLocation, leaseCollectionLocation, leaseHostOptions);`. Once you do that, you can `await` here and it should work.

It'd be good to do so we can remove this blocking call.